app/models/user.rb

```diff
@@ -1,7 +1,15 @@
 class User < ApplicationRecord
-  validates_presence_of :name, :email, :password_digest
+  validates_presence_of :name, :email, :password_digest, :api_key
+  validates_uniqueness_of :api_key
   validates :password_digest, confirmation: {case_sensitive: true}
   has_secure_password
   enum status: ['inactive', 'active']
 
+  before_validation :set_api_key
```

Comment:
So Josh left a message about how to refactor and avoid using a callback, but it also applies to the email piece. This works well enough for now. I am going to merge this in and we can refactor once email is also merged.